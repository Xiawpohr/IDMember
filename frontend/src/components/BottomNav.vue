<template>
  <v-bottom-nav app :value="isAuth">
    <v-btn
      color="teal"
      flat
      value="discover"
      to="/"
    >
      <span>Discover</span>
      <v-icon>whatshot</v-icon>
    </v-btn>
    <v-btn
      color="teal"
      flat
      value="friends"
      to="/friends"
    >
      <span>Friends</span>
      <v-icon>favorite</v-icon>
    </v-btn>
    <v-btn
      color="teal"
      flat
      value="profile"
      to="profile"
    >
      <span>Profile</span>
      <v-icon>person</v-icon>
    </v-btn>
  </v-bottom-nav>
</template>

<script>
export default {
  computed: {
    isAuth() {
      return this.$store.state.auth.id !== null
    }
  }
}
</script>

