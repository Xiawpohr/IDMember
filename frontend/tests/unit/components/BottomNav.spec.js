import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import BottomNav from '@/components/BottomNav.vue'

Vue.use(Vuetify)

it('should render the component', () => {
  const wrapper = shallowMount(BottomNav, { localVue: Vue })
  expect(wrapper.html()).toMatchSnapshot()
})
