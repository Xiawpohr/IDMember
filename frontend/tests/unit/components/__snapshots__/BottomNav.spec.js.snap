// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `
<vbottomnav-stub height="56" value="true" app="" color="transparent">
  <vbtn-stub activeclass="v-btn--active" flat="true" tag="button" type="button" value="discover" color="teal" to="/"><span>Discover</span>
    <vicon-stub>whatshot</vicon-stub>
  </vbtn-stub>
  <vbtn-stub activeclass="v-btn--active" flat="true" tag="button" type="button" value="friends" color="teal" to="/friends"><span>Friends</span>
    <vicon-stub>favorite</vicon-stub>
  </vbtn-stub>
  <vbtn-stub activeclass="v-btn--active" flat="true" tag="button" type="button" value="profile" color="teal" to="profile"><span>Profile</span>
    <vicon-stub>person</vicon-stub>
  </vbtn-stub>
</vbottomnav-stub>
`;
